shared/src/main/java/net/blay09/mods/waystones/menu/WarpPlateContainer.java

```diff
@@ -23,11 +23,36 @@ public WarpPlateContainer(int windowId, WarpPlateBlockEntity warpPlate, Containe
 
         checkContainerDataCount(containerData, 1);
 
-        addSlot(new Slot(warpPlate, 0, 80, 45));
-        addSlot(new Slot(warpPlate, 1, 80, 17));
-        addSlot(new Slot(warpPlate, 2, 108, 45));
-        addSlot(new Slot(warpPlate, 3, 80, 73));
-        addSlot(new Slot(warpPlate, 4, 52, 45));
+        addSlot(new Slot(warpPlate, 0, 80, 45) {
+            @Override
+            public boolean mayPickup(Player $$0) {
+                return warpPlate.isCompletedFirstAttunement() && super.mayPickup($$0);
+            }
+        });
+        addSlot(new Slot(warpPlate, 1, 80, 17) {
+            @Override
+            public boolean mayPickup(Player $$0) {
+                return warpPlate.isCompletedFirstAttunement() && super.mayPickup($$0);
+            }
+        });
+        addSlot(new Slot(warpPlate, 2, 108, 45) {
+            @Override
+            public boolean mayPickup(Player $$0) {
+                return warpPlate.isCompletedFirstAttunement() && super.mayPickup($$0);
+            }
+        });
+        addSlot(new Slot(warpPlate, 3, 80, 73) {
+            @Override
+            public boolean mayPickup(Player $$0) {
+                return warpPlate.isCompletedFirstAttunement() && super.mayPickup($$0);
+            }
+        });
+        addSlot(new Slot(warpPlate, 4, 52, 45) {
+            @Override
+            public boolean mayPickup(Player $$0) {
+                return warpPlate.isCompletedFirstAttunement() && super.mayPickup($$0);
+            }
+        });
```

Comment:
Would be better to have a separate Slot class that implements this check once instead of duplicating the code for each slot.